[Bug]: Garmin No Longer Connecting - ECONNREFUSED

### Are you willing to submit a PR?

_No response_

### What happened?

After one of the recent updates (latest Garmin sync was 29-Jul), there seems to be something that changed where it's not referencing the correct location for the Garmin container. I have it set up to use my IP:PORT (8017), but the error below looks like it's referencing the default port (8000). Nothing has changed in my compose files or settings and I have only been occasionally updating as new releases come out.

Error received:
```
[2026-08-07T12:55:16.966Z] [ERROR] Error fetching Garmin nutrition diary for user c8126769-73e1-42ca-8dc5-0ebeb4b86096 from 2026-07-08 to 2026-08-07: ECONNREFUSED
[2026-08-07T12:55:16.966Z] [ERROR] [garminService] Error during nutrition sync for user c8126769-73e1-42ca-8dc5-0ebeb4b86096: Error: Failed to fetch Garmin nutrition diary: connect ECONNREFUSED 172.18.0.3:8000
    at Object.fetchGarminNutritionDiary (/app/SparkyFitnessServer/integrations/garminconnect/garminConnectService.ts:457:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:104:5)
    at async Object.syncGarminData (/app/SparkyFitnessServer/services/garminService.ts:1300:27)
    at async <anonymous> (/app/SparkyFitnessServer/routes/garminRoutes.ts:497:22) {
  [cause]: AxiosError: connect ECONNREFUSED 172.18.0.3:8000
      at AxiosError.from (file:///app/SparkyFitnessServer/node_modules/.pnpm/axios@1.13.6/node_modules/axios/lib/core/AxiosError.js:7:24)
      at RedirectableRequest.handleRequestError (file:///app/SparkyFitnessServer/node_modules/.pnpm/axios@1.13.6/node_modules/axios/lib/adapters/http.js:866:27)
      at RedirectableRequest.emit (node:events:521:24)
      at eventHandlers.<computed> (/app/SparkyFitnessServer/node_modules/.pnpm/follow-redirects@1.15.11/node_modules/follow-redirects/index.js:49:24)
      at ClientRequest.emit (node:events:509:28)
      at emitErrorEvent (node:_http_client:109:11)
      at Socket.socketErrorListener (node:_http_client:593:5)
      at Socket.emit (node:events:509:28)
      at emitErrorNT (node:internal/streams/destroy:170:8)
      at emitErrorCloseNT (node:internal/streams/destroy:129:3)
      at Axios.request (file:///app/SparkyFitnessServer/node_modules/.pnpm/axios@1.13.6/node_modules/axios/lib/core/Axios.js:46:41)
      at process.processTicksAndRejections (node:internal/process/task_queues:104:5)
      at async Object.fetchGarminNutritionDiary (/app/SparkyFitnessServer/integrations/garminconnect/garminConnectService.ts:421:24)
      at async Object.syncGarminData (/app/SparkyFitnessServer/services/garminService.ts:1300:27)
      at async <anonymous> (/app/SparkyFitnessServer/routes/garminRoutes.ts:497:22) {
    isAxiosError: true,
    code: 'ECONNREFUSED',
    config: {
      transitional: [Object],
      adapter: [Array],
      transformRequest: [Array],
      transformResponse: [Array],
      timeout: 120000,
      xsrfCookieName: 'XSRF-TOKEN',
      xsrfHeaderName: 'X-XSRF-TOKEN',
      maxContentLength: -1,
      maxBodyLength: -1,
      env: [Object],
      validateStatus: [Function: validateStatus],
      headers: [Object [AxiosHeaders]],
      method: 'post',
      url: 'http://sparkyfitness-garmin:8000/data/nutrition_diary',
      data: '{"user_id":"c8126769-73e1-42ca-8dc5-0ebeb4b86096","tokens":"{\\"di_token\\":\\"eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImRpLW9hdXRoLXNpZ25lci1wcm9kLTIwMjQtcTEifQ.eyJtYW5hZ2VkX3N0YXR1cyI6IkRFRkFVTFQiLCJjbGllbnRfZWNvc3lzdGVtIjoiQ09OTkVDVCIsInNjb3BlIjpbIk9NVF9ESVNDT1VOVFNfUkVBRCIsIkRJX0NPQUNISU5HX1dSSVRFIiwiR0FSTUlOUEFZX1dSSVRFIiwiQVRQX1JFQUQiLCJHSFNfU0FNRCIsIklOU0lHSFRTX1JFQUQiLCJPTVRfRElTQ09VTlRTX0NSRUFURSIsIkNJUV9BUFBTVE9SRV9TRVJWSUNFU19DUkVBVEUiLCJESV9UT1VNX05BVElWRV9SRVBPUlRFRF9ERVRBSUxfV1JJVEUiLCJDT01NVU5JVFlfQ09VUlNFX1dSSVRFIiwiR0NPRkZFUl9XUklURSIsIkRJX0FQSV9SRUFEIiwiRFRfQ0xJRU5UX0FOQUxZVElDU19XUklURSIsIkNJUV9BUFBTVE9SRV9TRVJWSUNFU19ERUxFVEUiLCJPTVRfU1VCU0NSSVBUSU9OX1JFQUQiLCJESV9DT0FDSElOR19SRUFEIiwiQ09OTkVDVF9SRUFEIiwiQ09NTVVOSVRZX0NPVVJTRV9SRUFEIiwiR09MRl9BUElfUkVBRCIsIkdIU19VUExPQUQiLCJESVZFX0FQSV9SRUFEIiwiQ0lRX0FQUFNUT1JFX1NFUlZJQ0VTX1JFQUQiLCJDSVFfQVBQU1RPUkVfU0VSVklDRVNfVVBEQVRFIiwiQ09OTkVDVF9XUklURSIsIkNPTk5FQ1RfTUNUX0RBSUxZX0xPR19SRUFEIiwiR0FSTUlOUEFZX1JFQUQiLCJESV9PQVVUSF8yX0FVVEhPUklaQVRJT05fQ09ERV9DUkVBVEUiLCJHT0xGX0FQSV9XUklURSIsIklOU0lHSFRTX1dSSVRFIiwiUFJPRFVDVF9TRUFSQ0hfUkVBRCIsIk9NVF9DQU1QQUlHTl9SRUFEIiwiR0NPRkZFUl9SRUFEIiwiQVRQX1dSSVRFIl0sImlzcyI6Imh0dHBzOi8vZGlhdXRoLmdhcm1pbi5jb20iLCJtZmEiOnRydWUsInJldm9jYXRpb25fZWxpZ2liaWxpdHkiOlsiR0xPQkFMX1NJR05PVVQiLCJNQU5BR0VEX1NUQVRVUyJdLCJjbGllbnRfdHlwZSI6IlVOREVGSU5FRCIsImV4cCI6MTc4NTg5MzY2MiwiaWF0IjoxNzg1ODAxNjAwLCJnYXJtaW5fZ3VpZCI6IjVlYTAxMTkzLTdlNjktNGYwNC1hMmZlLTYwOTViZTJlNGJiYiIsImp0aSI6ImVhNGIyN2EyLTViMmItNGU3My1iOWRiLWM1ODcxOGUwZWY0MSIsImNsaWVudF9pZCI6IkdBUk1JTl9DT05ORUNUX01PQklMRV9BTkRST0lEX0RJXzIwMjVRMiJ9.Rw3_qHeErty3LLuk7L7lgvx55Tn-k-FSANjNZoFZhTHSfFGe6RmpwQzzAZttpoOcrANIRsnmGaWgm1KoRJKjIgHZfDWyVvk0L0MHaA3N5RhFsmndJZWuUo5bkwJOTwBTcw6g1yd0M6KnwsN7lCjs9WRw_khqk6mw4oKr4Nj-UvTlo9JElOlVWOgEhlhHrSvVcmzoa6DgceDUFS2ymfMC2-m5C5Ds5mAUX4etlZ-YVFIV_ewbcLrM4ufuwHNQggb3AtvsYcJDM7NTx7QLzgf-iP8jt8dzbbmUJ5-tV5PBleSfWV8mwpYyYRF-4nDV5PdtPyeqR3Pol9Vb2cMEUiPCkQ\\",\\"di_refresh_token\\":\\"eyJyZWZyZXNoVG9rZW5WYWx1ZSI6IjRkY2U2YzRhLWVlNTUtNGEyYS04M2RiLTJmYzgyMTQ0NDE4NiIsImdhcm1pbkd1aWQiOiI1ZWEwMTE5My03ZTY5LTRmMDQtYTJmZS02MDk1YmUyZTRiYmIifQ==\\",\\"di_client_id\\":\\"GARMIN_CONNECT_MOBILE_ANDROID_DI_2025Q2\\"}","start_date":"2026-07-08","end_date":"2026-07-14"}',
      allowAbsoluteUrls: true
    },
    request: Writable {
      _events: [Object],
      _writableState: [WritableState],
      _maxListeners: undefined,
      _options: [Object],
      _ended: true,
      _ending: true,
      _redirectCount: 0,
      _redirects: [],
      _requestBodyLength: 2240,
      _requestBodyBuffers: [Array],
      _eventsCount: 3,
      _onNativeResponse: [Function (anonymous)],
      _currentRequest: [ClientRequest],
      _currentUrl: 'http://sparkyfitness-garmin:8000/data/nutrition_diary',
      _timeout: null,
      Symbol(shapeMode): true,
      Symbol(kCapture): false
    },
    cause: Error: connect ECONNREFUSED 172.18.0.3:8000
        at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1706:16) {
      errno: -111,
      code: 'ECONNREFUSED',
      syscall: 'connect',
      address: '172.18.0.3',
      port: 8000
    }
  }
}```

### Steps to reproduce

Any time trying to sync garmin.

### Expected behavior

Should connect to garmin container and sync

### How are you running SparkyFitness?

Docker Compose

### SparkyFitness version

v1.6.1

### Which area is affected?

Integrations (Garmin, Fitbit, etc.)

### Docker Compose & Environment Configuration

```yaml

```

### Runtime Logs (Server, Web, or Mobile)

```shell

```

### Screenshots or Screen Recordings

_No response_

### Anything else?

_No response_

--- a/docker/docker-compose.dev.yml
+++ b/docker/docker-compose.dev.yml
@@ -51,7 +51,7 @@ services:
       SPARKY_FITNESS_EMAIL_USER: ${SPARKY_FITNESS_EMAIL_USER}
       SPARKY_FITNESS_EMAIL_PASS: ${SPARKY_FITNESS_EMAIL_PASS}
       SPARKY_FITNESS_EMAIL_FROM: ${SPARKY_FITNESS_EMAIL_FROM}
-      GARMIN_MICROSERVICE_URL: http://sparkyfitness-garmin:8000 # Add Garmin microservice URL
+      GARMIN_MICROSERVICE_URL: ${GARMIN_MICROSERVICE_URL:-http://sparkyfitness-garmin:${GARMIN_SERVICE_PORT:-8000}} # Garmin microservice URL
     volumes:
       - ../SparkyFitnessServer:/app/SparkyFitnessServer:z
       - /app/SparkyFitnessServer/node_modules

--- a/docker/docker-compose.prod.yml
+++ b/docker/docker-compose.prod.yml
@@ -47,7 +47,7 @@ services:
       SPARKY_FITNESS_EMAIL_USER: ${SPARKY_FITNESS_EMAIL_USER}
       SPARKY_FITNESS_EMAIL_PASS: ${SPARKY_FITNESS_EMAIL_PASS}
       SPARKY_FITNESS_EMAIL_FROM: ${SPARKY_FITNESS_EMAIL_FROM}
-      GARMIN_MICROSERVICE_URL: http://sparkyfitness-garmin:8000 # Add Garmin microservice URL
+      GARMIN_MICROSERVICE_URL: ${GARMIN_MICROSERVICE_URL:-http://sparkyfitness-garmin:${GARMIN_SERVICE_PORT:-8000}} # Garmin microservice URL
       HTTP_PROXY: ${HTTP_PROXY:-} # Optional forward proxy for the server's outbound requests
       HTTPS_PROXY: ${HTTPS_PROXY:-}
       NO_PROXY: ${NO_PROXY:-}

